plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakePartitionsTable.java

```diff
@@ -0,0 +1,216 @@
+import io.trino.spi.connector.ConnectorTableMetadata;
+import io.trino.spi.connector.ConnectorTransactionHandle;
+import io.trino.spi.connector.EmptyPageSource;
+import io.trino.spi.connector.FixedPageSource;
+import io.trino.spi.connector.SchemaTableName;
+import io.trino.spi.connector.SystemTable;
+import io.trino.spi.predicate.TupleDomain;
+import io.trino.spi.type.RowType;
+import io.trino.spi.type.Type;
+import io.trino.spi.type.TypeManager;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import static com.google.common.base.Predicates.alwaysTrue;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static com.google.common.collect.ImmutableMap.toImmutableMap;
+import static io.trino.plugin.deltalake.DeltaLakeErrorCode.DELTA_LAKE_INVALID_SCHEMA;
+import static io.trino.plugin.deltalake.transactionlog.DeltaLakeSchemaSupport.extractPartitionColumns;
+import static io.trino.spi.block.RowValueBuilder.buildRowValue;
+import static io.trino.spi.type.BigintType.BIGINT;
+import static io.trino.spi.type.TypeUtils.writeNativeValue;
+import static java.util.Objects.requireNonNull;
+
+public class DeltaLakePartitionsTable
+        implements SystemTable
+{
+    private final SchemaTableName tableName;
+    private final String tableLocation;
+    private final TransactionLogAccess transactionLogAccess;
+    private final TypeManager typeManager;
+    private final ConnectorTableMetadata tableMetadata;
+    private final List<DeltaLakeColumnHandle> partitionColumns;
+    private final List<RowType.Field> partitionFields;
+
+    public DeltaLakePartitionsTable(
+            ConnectorSession session,
+            SchemaTableName tableName,
+            String tableLocation,
+            TransactionLogAccess transactionLogAccess,
+            TypeManager typeManager)
+    {
+        this.tableName = requireNonNull(tableName, "tableName is null");
+        this.tableLocation = requireNonNull(tableLocation, "tableLocation is null");
+        this.transactionLogAccess = requireNonNull(transactionLogAccess, "transactionLogAccess is null");
+        this.typeManager = requireNonNull(typeManager, "typeManager is null");
+
+        this.partitionColumns = getPartitionColumns(session, new SchemaTableName(tableName.getSchemaName(), DeltaLakeTableName.tableNameFrom(tableName.getTableName())), this.tableLocation);
+
+        this.partitionFields = this.partitionColumns.stream()
+                .map(column -> RowType.field(column.getBaseColumnName(), column.getType()))
+                .collect(toImmutableList());
+
+        ImmutableList.Builder<ColumnMetadata> columnMetadataBuilder = ImmutableList.builder();
+
+        if (!this.partitionFields.isEmpty()) {
+            columnMetadataBuilder.add(new ColumnMetadata("partition", RowType.from(this.partitionFields)));
+        }
```

Comment:
I'd rather have the "partition" column  - even in this case - the table will be empty.

Can you pls do in preparatory commit the same for Iceberg - to have consistency for these table formats?

Reply:
I don't think it's possible to have empty Row type. Do you have suggestions on what type to make empty `partition`?

Reply:
Nevermind.